src/app.js

```diff
@@ -93,21 +95,34 @@ i18next
 
 app.use(i18nextMiddleware.handle(i18next));
 
+// Generate a new session ID asynchronously if needed
+// `express-session` does not support sync session ID generation
+// https://github.com/expressjs/session/issues/107
+app.use(async (req, res, next) => {
+  const sessionId = getCookie(req, SESSION_COOKIE_NAME, SESSION_SECRET);
+  if (!sessionId) {
+    req.generatedSessionId = await uid(24);
+  }
+  next();
+});
+
 app.use(
   session({
-    name: "ipv_core_service_session",
+    name: SESSION_COOKIE_NAME,
     store: sessionStore,
     saveUninitialized: false,
     secret: SESSION_SECRET,
     unset: "destroy",
     resave: false,
     cookie: {
-      name: "ipv_core_service_session",
+      name: SESSION_COOKIE_NAME,
       expires: false,
       secret: SESSION_SECRET,
       signed: true,
       secure: "auto",
     },
+    // Use the generated session ID, or fall back to the default behaviour
+    genid: (req) => req.generatedSessionId || uid.sync(24),
```

Comment:
Still need the fallback if e.g. the existing session id is invalid, but this should be infrequent